File: Data Structures & Algorithms/max-area-of-island/submission-0.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        rows, cols = len(grid), len(grid[0])
        res = 0
        directions = [[1, 0], [-1, 0], [0, 1], [0, -1]]
        self.count = 0
        def dfs(r, c):
            if r < 0 or c < 0 or r >= rows or c >= cols or grid[r][c] != 1:
                return 
            grid[r][c] = 0
            
            self.count += 1
            for dr, dc in directions:
                dfs(r + dr, c + dc)
            return 

        for r in range(rows):
            for c in range(cols):
                dfs(r, c)
                res = max(res, self.count)
                self.count = 0

        return res

File: Data Structures & Algorithms/valid-sudoku/submission-0.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {

        for(int i=0; i<board.size(); i++){
            unordered_set<int> rowCount, colCount;
            for(int j=0; j<board.size(); j++){
                bool rowS=true, colS=true;
                if(board[i][j] != '.'){
                    int val = board[i][j];
                    auto rowResult = rowCount.insert(val);
                    rowS = rowResult.second;
                }
                if(board[j][i] != '.'){
                    int val = board[j][i];
                    auto colResult = colCount.insert(val);
                    colS = colResult.second;
                }
                if(!colS || !rowS)
                    return false;
            }
        }

        for(int i=0; i<9; i+=3){
            for(int j=0; j<9; j+=3){
                unordered_set<int> count;
                for(int k=i; k<i+3; k++){
                    for(int l=j; l<j+3; l++){
                        if(board[k][l]!='.'){
                            int val = board[k][l];
                            auto res = count.insert(val);
                            if(!res.second)return false;
                        }
                    }
                }
            }
        }

        return true;
    }

};
